### PR TITLE
ovmf: add support for 4MB flash image

### PR DESCRIPTION
**Pull request by adamcstephens**
###### Description of changes

Adds support for building flash images targets of 2MB or 4MB. LXD expects both to be available, so I extracted them both to arguments. I'm happy to discuss alternative implementations.

This is in preparation for the [LXD VM work](https://github.com/NixOS/nixpkgs/pull/244093), but here's how I'm currently using this: https://github.com/NixOS/nixpkgs/blob/ef59bba98cbf698f1c6ba5fd4d012ef27c24a41d/pkgs/tools/admin/lxd/wrapper.nix#L58-L92

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by adamcstephens** [comment]
@RaitoBezarius Any thoughts on this?

File: pkgs/applications/virtualization/OVMF/default.nix [hunk 1]
```diff
@@ -1,5 +1,7 @@
 { stdenv, nixosTests, lib, edk2, util-linux, nasm, acpica-tools, llvmPackages
 , csmSupport ? false, seabios ? null
+, fdSize2MB ? csmSupport
```
Comment (RaitoBezarius) [inline pkgs/applications/virtualization/OVMF/default.nix:3]: Not sure why fdSize2MB defaults on csmSupport

Comment (RaitoBezarius) [reply pkgs/applications/virtualization/OVMF/default.nix:3]: Ah yes I see

File: pkgs/applications/virtualization/OVMF/default.nix [hunk 2]
```diff
@@ -100,6 +104,6 @@ edk2.mkDerivation projectDscPath (finalAttrs: {
     homepage = "https://github.com/tianocore/tianocore.github.io/wiki/OVMF";
     license = lib.licenses.bsd2;
     inherit (edk2.meta) platforms;
-    maintainers = [ lib.maintainers.raitobezarius ];
+    maintainers = [ lib.maintainers.adamcstephens lib.maintainers.raitobezarius ];
```
Comment (RaitoBezarius) [inline pkgs/applications/virtualization/OVMF/default.nix:107]: Please add another commit for the maintenance

Comment (SuperSandro2000) [reply pkgs/applications/virtualization/OVMF/default.nix:107]:
```suggestion
    maintainers = with lib.maintainers; [ adamcstephens raitobezarius ];
```

Comment (adamcstephens) [reply pkgs/applications/virtualization/OVMF/default.nix]: After formatting, maybe we can bless a linter too :)

File: pkgs/applications/virtualization/OVMF/default.nix (hunk 1 above)
Comment (adamcstephens) [reply pkgs/applications/virtualization/OVMF/default.nix:3]: I didn't dig into why it was already this way, but yeah this keeps the existing output the same for CSM.

**Review by RaitoBezarius — APPROVED** [review]

**Pull request merged**